src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2374,7 +2374,14 @@ private BoundExpression BindExplicitNullableCastFromNonNullable(ExpressionSyntax
                 var underlyingExpr = BindCastCore(node, operand, underlyingTargetTypeWithAnnotations, wasCompilerGenerated: false, diagnostics: bag);
                 if (underlyingExpr.HasErrors || bag.HasAnyErrors())
                 {
-                    Error(diagnostics, ErrorCode.ERR_NoExplicitConv, node, operand.Type, targetTypeWithAnnotations.Type);
+                    if (underlyingConversion.IsNumeric)
+                    {
+                        Error(diagnostics, ErrorCode.ERR_ConstOutOfRangeChecked, node, operand.ConstantValue.Value, targetTypeWithAnnotations.Type);
```

Comment:
Could we provide the operand's symbol if present? That way if the operand is something like `double.MaxValue` we can show that instead of `1.7976931348623157E+308`.

Reply:
I see that this is on par with the experience today for conversions not involving `System.Nullable`.

```cs
class C {
    int x = (int)long.MaxValue; // error CS0221: Constant value '9223372036854775807' cannot be converted to a 'int' (use 'unchecked' syntax to override)
}
```